iterreg/sparse/solvers.py

```diff
@@ -194,12 +194,16 @@ def cd(X, y, alpha, prox=shrink, pen=ell1, max_iter=1_000,
             all_w[t // f_store] = w
             if verbose:
                 print(t, E[t // f_store])
-
+            if ES and (t > 0):
+                if np.abs(E[t // f_store - 1]-E[t // f_store]) < 1e-10 * E[0]:
+                    E[(t // f_store):] = E[t // f_store]
+                    all_w[(t // f_store):] = w
```

Comment:
sure that this belongs to the stopping criterion check ? it was set L194

Reply:
It's different because I had the '":"
It's to set the remaining value to the last found with early stopping

Reply:
instead return all_w[:t//f_store + 1] in the end